node/overseer/src/lib.rs

```diff
@@ -1402,7 +1413,9 @@ where
 
 		self.clean_up_external_listeners();
 
-		self.broadcast_signal(OverseerSignal::ActiveLeaves(update)).await?;
+		if !update.is_empty() {
```

Comment:
this could be empty if we import the same block twice, how can this happen?
maybe we should log it

Reply:
That is actually a good question. I will take another look at the code why we see empty updates at all.

Reply:
You were right and after looking at the code, it was clear that the finalization function was the problem. We will very likely already have all leaves closed when they are finalized, leading to empty updates.